profiling/DerSPMF_test.jl

```diff
@@ -0,0 +1,70 @@
+using NonlinearEigenproblems, Random, SparseArrays, Revise, LinearAlgebra, BenchmarkTools
+import ..NEPTypes.AbstractSPMF
+struct DerSPMF{T<:AbstractMatrix,FDtype} <: AbstractSPMF{T}
+    spmf::SPMF_NEP{T}
+    fD::Matrix{FDtype}
+    σ::Number
```

Comment:
`Number` is not a concrete type so you will not get type stability. You probably need it the type parameters (but not in the constructor).